src/prop.c

```diff
@@ -355,6 +355,14 @@ SEXP prop_set_(SEXP object, SEXP name, SEXP check_sexp, SEXP value) {
   Rboolean should_validate_obj = check;
   Rboolean should_validate_prop = check;
 
+  // if already flagged for non-recursion, do not validate as this will be
+  // validated after top-level return
+  SEXP no_recurse_list = Rf_getAttrib(object, sym_dot_setting_prop);
+  if (pairlist_contains(no_recurse_list, name_sym)) {
+    should_validate_obj = FALSE;
+    should_validate_prop = FALSE;
+  }
```

Comment:
This is necessary for `new_object` calls that initialize properties using setters. Previously, when a setter would contain a call to `prop()<-`, this call's `check = TRUE` would force immediate validation.

Now, even if `check = TRUE`, we also look at this recursive property setting flag to see if we should ignore it.